### ISSUE TITLE
Empty paragraphs messing up CSS numbering

### ISSUE THREAD

**Issue by guicapanema** [GetmeUK/ContentTools#398]
Hello there! First of all thanks for making ContentTools -- it's pretty awesome.

I'm developing a personal project where I'd like my paragraphs to be numbered (I'm using CSS counters for that). Mostly everything works fine, except for an edge case where an empty paragraph is left when saving a document:

In editor view:
<img width="964" alt="captura de tela 2017-03-26 as 11 38 58" src="https://cloud.githubusercontent.com/assets/8330306/24332199/e532dcca-1218-11e7-9e80-e48753279311.png">

After saving:
<img width="960" alt="captura de tela 2017-03-26 as 11 39 12" src="https://cloud.githubusercontent.com/assets/8330306/24332198/e52d4256-1218-11e7-9f52-c57a4d9b8046.png">

I'm using the sample sandbox code and tried adding `editor._emptyRegionsAllowed = false;` to it. Here's how I'm initializing the editor:

	
	window.onload = function() {
	var FIXTURE_TOOLS, editor, req;
	ContentTools.IMAGE_UPLOADER = cloudinaryImageUploader;
	ContentTools.StylePalette.add([new ContentTools.Style('By-line', 'article__by-line', ['p']), new ContentTools.Style('Caption', 'article__caption', ['p']), new ContentTools.Style('Example', 'example', ['pre']), new ContentTools.Style('Example + Good', 'example--good', ['pre']), new ContentTools.Style('Example + Bad', 'example--bad', ['pre'])]);
	editor = ContentTools.EditorApp.get();
	
	editor.init('[data-editable], [data-fixture]', 'data-name');
	editor._emptyRegionsAllowed = false;

Etc. Am I doing something wrong?

Thanks again!
Guilherme

**Comment by anthonyjb** [comment]
Hi @guicapanema - that's an interesting one - could you send me a snapshot of the page HTML from the DOM/element inspector after the save when this glitch is visible.

Re. `_emptyRegionsAllowed` I don't believe setting this would help in this situation as it's to do with the whole region being empty not a single paragraph.

**Comment by guicapanema** [comment]
Hi @anthonyjb ! Thanks for answering and sorry for the delay.

This is what the HTML looks like:
```
<div class="section" id="conclusion" data-editable="" data-name="conclusion"><h2 class="section-title">
    Conclusão
</h2>
<p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
</p>
<p>
    
</p>
<p>
    Morbi aliquet leo sit amet nisl luctus facilisis sit amet sit amet eros.
</p>
</div>
```

And this is the result:

<img width="506" alt="captura de tela 2017-05-01 as 09 36 45" src="https://cloud.githubusercontent.com/assets/8330306/25579819/bbd93b1c-2e51-11e7-8fe9-e1acc0352ac6.png">

From what I see the problem is the empty `<p>` tag is left in the DOM. It has the CSS's `:before` attached to it, which is what messes up the numbering. Not a huge deal, but should be an easy fix, right?

**Comment by anthonyjb** [comment]
@guicapanema so you can remove that behaviour by changing the CSS of course however the default CSS uses an :after pseudo element for empty elements, see here: https://github.com/GetmeUK/ContentEdit/blob/master/src/styles/content-edit.scss#L164

So the before bit is confusing me a little bit.

**Comment by guicapanema** [comment]
Hi @anthonyjb . Actually, I didn't explain myself well. I added the :before element because I need paragraphs to be numbered, and that's where I'm putting my counters. In a normal situation (that is, with no empty <p> tags), that should work fine.

The main thing here is that I would like the editor to remove empty paragraphs before saving to the database. When I reported this, I was thinking no one would like empty <p>s, but that was really shortsighted.

Perhaps adding an option?
